python/opae.admin/opae/admin/config.py

```diff
@@ -0,0 +1,579 @@
+        print(f'Error parsing config: "devices" key not a list.')
+        return (False, fpgareg_cfg)
+
+    devs = parse_devices(c[key])
+    if not devs:
+        print(f'Error parsing config: "devices" is empty.')
+        return (False, fpgareg_cfg)
+
+    key = 'opae'
+    if key not in c or type(c[key]) != dict:
+        print(f'Error parsing config: "opae" key '
+              f'missing or invalid in {config}.')
+        return (False, fpgareg_cfg)
+
+    opae = c[key]
+    key = 'fpgareg'
+    if key not in opae or type(opae[key]) != list:
+        print(f'Warning parsing config: "fpgareg" key '
+              f'missing or invalid in {config} "opae".')
+        return (True, fpgareg_cfg) # continue parsing
+
+    value = { 'platform': platform }
+
+    fpgareg = opae[key]
+    for f in fpgareg:
+        key = 'enabled'
+        if key not in f or not f[key]:
+            continue
+
+        key = 'devices'
+        if key not in f or len(f[key]) == 0:
+            print(f'Warning parsing config: "devices" key '
+                  f'missing or invalid in {config} "fpgareg".')
+            continue
+
+        for d in f['devices']:
+            if d not in devs:
+                print(f'Error parsing config: {d} not found '
+                      f'in "devices" for {config}.')
+                return (False, fpgareg_cfg)
+    
+            fpgareg_cfg[devs[d]] = value
+
+    return (True, fpgareg_cfg)
+
+
+def load_fpgareg_configuration(cfg):
+    """Walk through each "configs" section of the given parsed
+       JSON object cfg. If cfg is None or if an error is encountered
+       during parsing, then return None. Otherwise, return the parsed
+       fpgareg object.
+    """
+    if not cfg: # No parsed config from file.
+        return None
+
+    if not 'configs' in cfg:
+        print(f'Error parsing config: "configs" key not found.')
+        return None
+    if not 'configurations' in cfg:
+        print(f'Error parsing config: "configurations" key not found.')
+        return None
+
+    fpgareg_cfg = {}
+    for c in cfg['configs']:
+        status, fpgareg_cfg = load_fpgareg_config(fpgareg_cfg,
+                                                  cfg,
+                                                  c,
+                                                  cfg['configurations'])
+        if not status:
+            return None
+
+    return fpgareg_cfg
+
+
+def print_fpgareg_configuration(cfg):
+    """Pretty print an fpgareg configuration data object.
+    """
+    for key in cfg:
+        print(f'0x{key[0]:04x}:0x{key[1]:04x} ', end='')
+        if key[2] == OPAE_VENDOR_ANY:
+            print(f'*     ', end='')
+        else:
+            print(f'0x{key[2]:04x}', end='')
+        print(':', end='')
+        if key[3] == OPAE_DEVICE_ANY:
+            print(f'*     ', end='')
+        else:
+            print(f'0x{key[3]:04x}', end='')
+
+        print(' ', end='')
+        print(cfg[key])
+
+
+def load_configs():
```

Comment:
Just trying to control the noise..  If we let users pass in the config file as an argument, then the system can become cluttered with config files.  If we leave it like this, we have a set of places we can look that are actually manageable in case we need to manually clean them when troubleshooting.